### PR TITLE
switch to using --check

### PR DESCRIPTION
**Pull request by gavinuhma**
Implementing the suggestions from https://news.ycombinator.com/item?id=33377273

### CODE REVIEW

File: checksum.sh [hunk 1]
```diff
@@ -1,18 +1,17 @@
 #!/bin/bash
 
 function checksum() {
+  local s
   s=$(curl -fsSL "$1")
   if ! command -v shasum >/dev/null
   then
     shasum() { sha1sum "$@"; }
```
Comment (dundarious) [inline checksum.sh:8]: sha1sum doesn't support the `-a` option you always give below. You just shouldn't pass `-a`. I only used it when contriving an example (where I had to mock having something to download which has a hash already).

And you can just use a variable for the command (you don't need `eval` or anything like that to do such things).

```
local hasher
if command -v shasum >/dev/null ; then
  hasher=shasum
else
  hasher=sha1sum
fi
...
printf %s\\n "$s" | "$hasher" --check --status ...
```

File: checksum.sh [hunk 2]
```diff
@@ -1,18 +1,17 @@
 #!/bin/bash
 
 function checksum() {
+  local s
   s=$(curl -fsSL "$1")
```
[the remark this answers is not in the document]
Comment (gavinuhma) [reply checksum.sh:5]: > the shell variable capture is probably non-viable.

@dundarious given that, downloading the script to a tmp file may be the best solution

[the remark this answers is not in the document]
Comment (gavinuhma) [reply checksum.sh:5]: https://github.com/gavinuhma/checksum.sh/issues/6

File: checksum.sh (hunk 1 above)
Comment (gavinuhma) [reply checksum.sh]: Updated. Great suggestion. Thanks!

**Pull request merged**